src/gui/editors/PianoRoll.cpp

```diff
@@ -976,82 +976,88 @@ void PianoRoll::drawNoteRect( QPainter & p, int x, int y,
 
 
 
-void PianoRoll::drawDetuningInfo( QPainter & _p, const Note * _n, int _x,
-								int _y ) const
+void PianoRoll::drawDetuningInfo(QPainter & p, const Note * n, int x, int y) const
 {
-	int middle_y = _y + m_keyLineHeight / 2;
-	_p.setPen(m_noteColor);
-	_p.setClipRect(
+	// Get a timeMap of the detuning pattern
+	AutomationPattern* detuningPattern = n->detuning()->automationPattern();
+	timeMap & detuningMap = detuningPattern->getTimeMap();
+
+	// Return if it's empty
+	if (detuningMap.isEmpty()) { return; }
+
+	// Set the detuning color to the same color as the notes,
+	// but lighter and with some alpha
+	QColor detuningColor = QColor(m_noteColor.lighter(150));
+	detuningColor.setAlpha(70);
+
+	// Get the first node
+	timeMap::iterator it = detuningMap.begin();
+
+	// Set the clip area to the PianoRoll area
+	p.setClipRect(
 		m_whiteKeyWidth,
 		PR_TOP_MARGIN,
 		width() - m_whiteKeyWidth,
-		keyAreaBottom() - PR_TOP_MARGIN);
+		keyAreaBottom() - PR_TOP_MARGIN
```

Comment:
```suggestion
		keyAreaBottom() - keyAreaTop()
```

Reply:
Done! We should soon open a PR and run some `sed` to replace it. PR_TOP_MARGIN is still being used in lots of places (including the original drawDetuningInfo method).

Reply:
We can update it as it comes along. I'd rather not worry about that level of detail just yet.